api/migrations/versions/87213d612eaf_eeml_application_form.py

```diff
@@ -0,0 +1,327 @@
+
+    def __init__(self, application_form_id, name, description, order):
+        self.application_form_id = application_form_id
+        self.name = name
+        self.description = description
+        self.order = order
+
+class Question(Base):
+    __tablename__ = 'question'
+    __table_args__ = {'extend_existing': True}
+
+    id = db.Column(db.Integer(), primary_key=True)
+    application_form_id = db.Column(db.Integer(), db.ForeignKey('application_form.id'), nullable=False)
+    section_id = db.Column(db.Integer(), db.ForeignKey('section.id'), nullable=False)
+    type = db.Column(db.String(), nullable=False)
+    description = db.Column(db.String(), nullable=True)
+    headline = db.Column(db.String(), nullable=False)
+    placeholder = db.Column(db.String(), nullable=True)
+    validation_regex = db.Column(db.String(), nullable=True)
+    validation_text = db.Column(db.String(), nullable=True)
+    order = db.Column(db.Integer(), nullable=False)
+    options = db.Column(db.JSON(), nullable=True)
+    is_required = db.Column(db.Boolean(), nullable=False)
+
+    def __init__(self, application_form_id, section_id, headline, placeholder, order, questionType, validation_regex, validation_text=None, is_required = True, description = None, options = None):
+        self.application_form_id = application_form_id
+        self.section_id = section_id
+        self.headline = headline
+        self.placeholder = placeholder
+        self.order = order
+        self.type = questionType
+        self.description = description
+        self.options = options
+        self.is_required = is_required
+        self.validation_regex = validation_regex
+        self.validation_text = validation_text
+
+
+def upgrade():
+    Base.metadata.bind = op.get_bind()
+    session = orm.Session(bind=Base.metadata.bind)
+
+    # Reset auto-increment ids
+    op.get_bind().execute("""SELECT setval('event_id_seq', (SELECT max(id) FROM event));""")
+    op.get_bind().execute("""SELECT setval('application_form_id_seq', (SELECT max(id) FROM application_form));""")
+    op.get_bind().execute("""SELECT setval('section_id_seq', (SELECT max(id) FROM section));""")
+    op.get_bind().execute("""SELECT setval('question_id_seq', (SELECT max(id) FROM question));""")
+
+    # Add event
+    eeml2020 = Event('EEML 2020', 'Eastern European Machine Learning Summer School 2020, 6-11 July 2020, Krakow, Poland',
+        datetime.date(2020, 7, 6), datetime.date(2020, 7, 11), 'eeml2020', 2, 'contact@eeml.eu',
+        'http://eeml.eu', datetime.date(2020, 2, 2), datetime.date(2020, 3, 20), datetime.date(2020, 3, 20), 
+        datetime.date(2020, 4, 20), datetime.date(2020, 4, 20), datetime.date(2020, 5, 20), datetime.date(2020, 5, 20),
+        datetime.date(2020, 6, 20), datetime.date(2020, 6, 20), datetime.date(2020, 6, 30))
+
+    session.add(eeml2020)
+    session.commit()
+
+    app_form = ApplicationForm(eeml2020.id, True, datetime.date(2020, 3, 20))
```

Comment:
Seems we have some duplication now in the ApplicationForm deadline field. This is the same as the application_close field in Event. Should we clean this up? We can maybe make a refactor issue?

Reply:
Have made an issue to clean this up (#518) - thanks :)